### DISCUSSION TITLE
dp convert successful but running Lammps gives error

### DISCUSSION THREAD

**Discussion by rajnichahal** [deepmodeling/deepmd-kit#1725] [Q&A]
Hi,
I trained the deepmd model using deepmd version 1.2.1. In order to use it for running Lammps simulations using deepmd version 2.1.0, I did the following:
dp convert-from 1.2 -i old_frozen_model.pb -o new_frozen_model.pb

The conversion was successful, but when I am running Lammps simulations using the new_frozen_model.pb in version 2.1.0, I am getting the following error:

2022-05-23 14:28:47.179610: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2 AVX AVX2 AVX512F FMA
2022-05-23 14:28:47.493488: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 1399990000 Hz
2022-05-23 14:28:47.555565: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x5555562aa980 initialized for platform Host (this does not guarantee that XLA will be used). Devices:
2022-05-23 14:28:47.556251: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version
Not found: Op type not registered 'ProdEnvMatA' in binary running on nid03436. Make sure the Op and Kernel are registered in the binary running in this process. Note that if you are loading a saved graph which used ops from tf.contrib, accessing (e.g.) `tf.contrib.resampler` should be done before importing the graph, as contrib ops are lazily registered when the module is first accessed.

Please let me know if you could help me resolve this issue. Thanks!

**Comment by njzjz** [comment]
You may load the wrong library. Set the environment variables `export LD_DEBUG=libs` to see which library was loaded.

**Comment by rajnichahal** [comment]
Hi, pardon my ignorance: but is setting the environment variable relevant when I am using the singularity image of deepmd?

For both training the potential (v 1.2.1) as well as using the trained potential in Lammps (v 2.1.0), I am using the pulled singularity images.

**Reply by njzjz** [reply]
What image did you use? What version did the program print? It looks that you did not use a correct version.

**Reply by rajnichahal** [reply]
Hi, I used following command to pull the image for dpkit version 2.1.0:
**/usr/common/das/shifter/pull ghcr.io/deepmodeling/deepmd-kit:2.1.0_cpu**

I believe somehow it is using the Lammps from versiob 1.2.1. Please take a look below:
Summary of lammps deepmd module ...
  >>> Info of deepmd-kit:
  installed to:       /opt/deepmd-kit
  **source:             v1.2.1-3-g30922e7**
  source brach:       HEAD
  source commit:      30922e7
  source commit at:   2020-09-14 22:49:07 +0800
  build float prec:   double
  build with tf inc:  /opt/deepmd-kit/include;/opt/deepmd-kit/include
  build with tf lib:  /opt/deepmd-kit/lib/libtensorflow_cc.so;/opt/deepmd-kit/lib/libtensorflow_framework.so
  set tf intra_op_parallelism_threads: 0
  set tf inter_op_parallelism_threads: 0
  >>> Info of lammps module:
  use deepmd-kit at:  /opt/deepmd-kit
  **source:             v1.2.1-3-g30922e7**

**Reply by njzjz** [reply]
I will investigate it.

**Reply by njzjz** [reply]
I cannot reproduce and I got the correct version:
```
  >>> Info of deepmd-kit:
  installed to:       /opt/deepmd-kit
  source:             v2.1.0
  source branch:       HEAD
  source commit:      3e54fea7
  source commit at:   2022-03-07 11:41:41 +0800
  surpport model ver.:1.1
  build float prec:   double
  build with tf inc:  /opt/deepmd-kit/include;/opt/deepmd-kit/include
  build with tf lib:  /opt/deepmd-kit/lib/libtensorflow_cc.so;/opt/deepmd-kit/lib/libtensorflow_framework.so
  set tf intra_op_parallelism_threads: 0
  set tf inter_op_parallelism_threads: 0
```

Maybe you need to check if you pull the correct image

**Reply by rajnichahal** [reply]
Thank you -- it is working now. I was loading previous deepmd version while running Lammps job.
